### PR TITLE
refactor(table): use file name instead package path

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Package path can be very long then table is a bit too wide.
For table used only package file name.
JSON result, as before, contains full package name .

Also added log message about using `--format json` subcommand to get full path to package.

Example:
```
➜  trivy git:(change_filepath_to_table_result) ✗ go run cmd/trivy/main.go rootfs /home/dmitriy/work/temp/issue1958/jars/v2/folder1
2022-04-08T16:14:40.225+0600    INFO    Number of language-specific files: 1
2022-04-08T16:14:40.225+0600    INFO    Detecting jar vulnerabilities...
2022-04-08T16:14:40.226+0600    INFO    Table result includes only package filenames. Use '--format json' subcommand to get full path to the package file.

Java (jar)
==========
Total: 1 (UNKNOWN: 0, LOW: 0, MEDIUM: 1, HIGH: 0, CRITICAL: 0)

+-------------------------------------+------------------+----------+-------------------+-----------------------+---------------------------------------+
|               LIBRARY               | VULNERABILITY ID | SEVERITY | INSTALLED VERSION |     FIXED VERSION     |                 TITLE                 |
+-------------------------------------+------------------+----------+-------------------+-----------------------+---------------------------------------+
| org.apache.logging.log4j:log4j-core | CVE-2021-44832   | MEDIUM   | 2.17.0            | 2.17.1, 2.12.4, 2.3.2 | log4j-core: remote code               |
| (log4j-core-2.17.0.jar)             |                  |          |                   |                       | execution via JDBC Appender           |
|                                     |                  |          |                   |                       | -->avd.aquasec.com/nvd/cve-2021-44832 |
+-------------------------------------+------------------+----------+-------------------+-----------------------+---------------------------------------+

```

## Related PRs
- [x] #aquasecurity/trivy/pull/1960

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy/blob/main/CONTRIBUTING.md) to this repository.
- [x] I've followed the [conventions](https://github.com/aquasecurity/trivy/blob/main/CONTRIBUTING.md#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [x] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/report/table.go [hunk 1]
```diff
@@ -4,16 +4,21 @@ import (
...
 	"github.com/olekukonko/tablewriter"
 	"golang.org/x/exp/slices"
 
 	dbTypes "github.com/aquasecurity/trivy-db/pkg/types"
+	"github.com/aquasecurity/trivy/pkg/log"
 	"github.com/aquasecurity/trivy/pkg/types"
 )
 
+// We have to show a message once about using the '-format json' subcommand to get the full pkgPath
+var filePathMessageShown = false
```
Comment (knqyf263) [inline pkg/report/table.go:20]: Once.Do may be good for this purpose.
https://pkg.go.dev/sync#Once.Do

File: pkg/report/table.go [hunk 2]
```diff
@@ -134,7 +140,11 @@ func (tw TableWriter) setVulnerabilityRows(table *tablewriter.Table, vulns []typ
 		severityCount[v.Severity]++
 		lib := v.PkgName
 		if v.PkgPath != "" {
-			lib = fmt.Sprintf("%s (%s)", v.PkgName, v.PkgPath)
+			fileName := filepath.Base(v.PkgPath)
+			lib = fmt.Sprintf("%s (%s)", v.PkgName, fileName)
+			showMessageOnce.Do(func() {
+				log.Logger.Infof("Table result includes only package filenames. Use '--format json' subcommand to get full path to the package file.")
```
Comment (knqyf263) [inline pkg/report/table.go:146]: Table result includes only package filenames. Use '--format json' option to get full path to the package file.

```suggestion
				log.Logger.Infof("Table result includes only package filenames. Use '--format json' option to get the full path to the package file.")
```

File: pkg/report/table.go [hunk 3]
```diff
@@ -4,16 +4,22 @@ import (
...
 	"github.com/olekukonko/tablewriter"
 	"golang.org/x/exp/slices"
 
 	dbTypes "github.com/aquasecurity/trivy-db/pkg/types"
+	"github.com/aquasecurity/trivy/pkg/log"
 	"github.com/aquasecurity/trivy/pkg/types"
 )
 
+// We have to show a message once about using the '-format json' subcommand to get the full pkgPath
+var showMessageOnce sync.Once
```
Comment (knqyf263) [inline pkg/report/table.go:21]: I think we should put it to `TableWriter` as this package is shared with other writers as well.

**Review by knqyf263 — CHANGES_REQUESTED** [review]

File: pkg/report/table.go (hunk 3 above)
Comment (DmitriyLewen) [reply pkg/report/table.go]: Done

File: pkg/report/table.go [hunk 4]
```diff
@@ -19,6 +22,9 @@ type TableWriter struct {
 	Severities []dbTypes.Severity
 	Output     io.Writer
 
+	// We have to show a message once about using the '-format json' subcommand to get the full pkgPath
+	ShowMessageOnce *sync.Once
```
Comment (knqyf263) [inline pkg/report/table.go:26]: Does it need to be a pointer? I expected something like this.

```suggestion
	showMessageOnce sync.Once
```

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**